mopidy/file/library.py

```diff
@@ -82,7 +82,14 @@ def browse(self, uri):
             elif os.path.isfile(child_path):
                 result.append(models.Ref.track(name=name, uri=uri))
 
-        result.sort(key=operator.attrgetter('name'))
+        sort_funcs = [operator.attrgetter('name'),
+                      lambda item: 0 if item.type == models.Ref.DIRECTORY
+                      else 1]
+        logger.debug('sort_funcs %s', sort_funcs)
+
+        for key_func in sort_funcs:
```

Comment:
Doing it like this I you will end up first sorting it just by name, and then resorting it just by `type == models.Ref.DIRECTORY`. On a side note `int(True) -> 1` and `int(False) -> 0`. With this in mind I think you want something like:

```
sorted([('directory', 'xyz'), ('directory', 'def'), ('directory', 'abc'), ('track', 'xyz'), ('track', 'def')], key=lambda v: (v[0] != 'directory', v[1]))

[('directory', 'abc'),
 ('directory', 'def'),
 ('directory', 'xyz'),
 ('track', 'def'),
 ('track', 'xyz')]
```

Which would give you `result.sort(sort=lambda ref: (ref.type != models.Ref.DIRECTORY, ref.name))` as the sort function you need.